common/src/commonMain/kotlin/com.harmony.kotlin/data/datasource/network/GenericNetworkDataSource.kt

```diff
@@ -98,48 +113,71 @@ open class GetNetworkDataSource<T>(
   override suspend fun getAll(query: Query): List<T> {
     return network {
       val response = when (query) {
+
+        is GenericOAuthNetworkQuery -> {
+          httpClient.get<String> {
+            oauthPasswordHeader(getPasswordTokenInteractor = query.getPasswordTokenInteractor)
+            createHttpRequestFromGenericNetworkQuery(query)
+          }
+        }
+
+        is GenericNetworkQuery -> {
+          httpClient.get {
+            createHttpRequestFromGenericNetworkQuery(query)
+          }
+        }
+
+        else -> {
+          httpClient.get<String>(_url)
+        }
```

Comment:
The resolution of this operation is exactly the same as the **get()** function because the only difference is the return type. 

Maybe we could have a private function to return the resolution of the HTTP properties (URL, params, headers) required for the http call. 

What do you think?

Reply:
I like your idea. If I understood your proposal we want a method or methods that resolve the URL, params and headers. Once we got all of them, use them in a final httpclient call.

Basically, the only difference is OAuth. if we need to add a header or not......

My question here is: do we want a single return with a Triple or we want a method for each? I prefer one method for each because it's more reusable.

Reply:
I'm not happy 100% but I think we have a good starting point. I would like to improve how we generate the request still.